fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java

```diff
@@ -387,7 +387,9 @@ public void commitPreparedTransaction(Database db, long transactionId, long time
             // so we just return false to indicate publish timeout
             throw new UserException("publish timeout: " + timeoutMillis);
         }
-        waiter.await(publishTimeoutMillis, TimeUnit.MILLISECONDS);
+        if (!waiter.await(publishTimeoutMillis, TimeUnit.MILLISECONDS)) {
+            throw new UserException("publish timeout: " + timeoutMillis);
+        }
```

Comment:
The Flink-connector will retry the ingestion again if the `sink.max-retries` is larger than 1. The retry will lead to another error like the `the label already exists`. I think it will also need to change the logic in the connector.

Reply:
talked with @banmoy offline, this will be another issue in flink-connector repo

Reply:
1. This publish timeout exception only trigger retry for `commit` operation, and `the label already exists` will not happen because we are not beginning a transaction. I check the connector's retry logic, and it will not be affected by this new exception.
2. But I think we should make it clear what status should be returned to the client if the transaction is committed but not published. In the current logic of `TransactionLoadAction#executeTransaction`,  a `FAILED` status will be returned to the client if adding this publish timeout exception,  and the client will retry `commit`.  Maybe we should still return `OK` status in this case. What do you think? @meegoo

Reply:
please note that before this PR, the code already throws publish timeout exception in some cases(in some case it won't), this PR only change the behavior to: if it timeout, make sure to throw an exception.

Reply:
>  a FAILED status will be returned to the client if adding this publish timeout exception, and the client will retry commit
retry `commit` is not needed, return OK is also problematic, a user should notice the txn is not complete yet(but does not need to retry)

Reply:
> please note that before this PR, the code already throws publish timeout exception in some cases(in some case it won't), this PR only change the behavior to: if it timeout, make sure to throw an exception.

You are right. Thanks for the reminding.

Reply:
> > a FAILED status will be returned to the client if adding this publish timeout exception, and the client will retry commit
> 
> retry `commit` is not needed, return OK is also problematic, a user should notice the txn is not complete yet(but does not need to retry)

Yes, maybe we need return a status `TStatusCode.PUBLISH_TIMEOUT` rather than `FAILED`. But it's not related to this PR.